frb_example/pure_dart_multi/dart/run.sh

```diff
@@ -2,15 +2,15 @@
 
 set -eux
 
-if [[ -z "${CARGO_TARGET_DIR}" ]]; then
-  echo 'Please set environment variable CARGO_TARGET_DIR'
-  exit 1
-fi
+# if [[ -z "${CARGO_TARGET_DIR}" ]]; then
+#   echo 'Please set environment variable CARGO_TARGET_DIR'
+#   exit 1
+# fi
```

Comment:
Temporary hack to check CI is ok, just remember not to submit to the final PR :)

Reply:
> Temporary hack to check CI is ok, just remember not to submit to the final PR :)

Actually, I am trying to figure out what make [this ](https://github.com/fzyzcjy/flutter_rust_bridge/runs/7174628203?check_suite_focus=true)issue;  

I still think there should be `api_1.rs` and `api_2.rs` in `pure_dart_multi\rust\src`, but not `api.rs` and `api_2.rs`.

Reply:
> I still think there should be api_1.rs and api_2.rs in pure_dart_multi\rust\src, but not api.rs and api_2.rs.

Sure, you can put those two names.

Btw it will be even greater if it is named with meaning, e.g. `api_apple`, `api_orange`, etc. Otherwise people may wrongly think only numbers are supported.

Reply:
>   thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/runner/work/flutter_rust_bridge/flutter_rust_bridge/frb_codegen/src/config.rs:379:77

What about print out some error message during this unwrap. For example, replace `.unwrap()` with `.expect("the file is: {}", your_file_name)`. Then you know which file is it thinking.

For example, it maybe because, when assembling filename, you add an extra `/` etc

Reply:
> > I still think there should be api_1.rs and api_2.rs in pure_dart_multi\rust\src, but not api.rs and api_2.rs.
> 
> Sure, you can put those two names.
> 
> Btw it will be even greater if it is named with meaning, e.g. `api_apple`, `api_orange`, etc. Otherwise people may wrongly think only numbers are supported.

I would give detailed explanation in doc afterwards.

Reply:
> > thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/runner/work/flutter_rust_bridge/flutter_rust_bridge/frb_codegen/src/config.rs:379:77
> 
> What about print out some error message during this unwrap. For example, replace `.unwrap()` with `.expect("the file is: {}", your_file_name)`. Then you know which file is it thinking.
> 
> For example, it maybe because, when assembling filename, you add an extra `/` etc

The [issue](https://github.com/fzyzcjy/flutter_rust_bridge/runs/7179679911?check_suite_focus=true) comes from that it still using `api.rs` instead of `api_1.rs` or `api_2.rs` as the path for `cargo:rerun-if-changed=...`. Is it related to something like cache, which makes it still using the old `cargo:rerun-if-changed=...` data when running `Run valgrind script (multi blocks)`?

Reply:
Maybe search what is `cargo:rerun-if-changed=` and detailed documentation will be seen

Reply:
> Maybe search what is `cargo:rerun-if-changed=` and detailed documentation will be seen

[It](https://github.com/fzyzcjy/flutter_rust_bridge/runs/7181080218?check_suite_focus=true) still found the wrong path, I guess there is something wrong with the CI statement.
```yaml
  - name: Run valgrind script (multi blocks)
    working-directory: ./frb_example/pure_dart_multi/dart
    run: chmod +x ./run_multi.sh && ./run_multi.sh
    env:
      CARGO_TARGET_DIR: /home/runner # NOTE here, must be an absolute path, since in my script uses it in multiple different directories
```
How do you think?

Reply:
What is the correct path?

Reply:
> What is the correct path?

Let's take a look at this [issue](https://github.com/fzyzcjy/flutter_rust_bridge/runs/7188985333?check_suite_focus=true),
```
  --- stderr
  thread 'main' panicked at 'panic with file: /home/runner/work/flutter_rust_bridge/flutter_rust_bridge/frb_example/pure_dart_multi/rust/src/api.rs', 
```
There is no `api.rs`, only `api_1.rs` and `api_2.rs`. And for testing, I even removed statement like `println!("cargo:rerun-if-changed={}", file_name);`
from `/pure_dart_multi/rust/build.rs`.

So I wonder what makes this issue occur.

Reply:
Maybe dig into the rust code which generates such wrong path.

Reply:
Btw you can also debug locally by running those commands on your computer

Reply:
> Btw you can also debug locally by running those commands on your computer

Now I am pretty sure it is caused by the shared `flutter_rust_bridge_example`. Look at [here](https://github.com/fzyzcjy/flutter_rust_bridge/runs/7189850597?check_suite_focus=true):  I switch the order of single and multi blocks, and as expected, the latter one is ruined, while the former one is ok.  And in this case, they both used the same `flutter_rust_bridge_example-3b9163c44a994e76`.

But... how to fix it?

Reply:
Then, instead of one `valgrind script` in CI, lets create two:

* valgrind script (the old one)
* pure_dart_multi

Then, the two runs in parallel without any interference

Reply:
> Then, instead of one `valgrind script` in CI, lets create two "job"s:
> 
> * valgrind script (the old one)
> * pure_dart_multi
> 
> Then, the two runs in parallel without any interference
> 
> Btw we already have a ton of jobs. For example, valgrind test, run codegen, ...


I find a better way. The `[lib]` from `Cargo.toml` used by single and multi blocks examples are now separated by different names `flutter_rust_bridge_example` and `flutter_rust_bridge_example_multi`. Now there is [no valgrind conflict issue](https://github.com/fzyzcjy/flutter_rust_bridge/runs/7193628108?check_suite_focus=true).   

By the way, to make distinguishable,https://github.com/fzyzcjy/flutter_rust_bridge/pull/527/commits/c6ed19d4d2501d472e0891fce274b9662d5e38be, I also renamed the package names with `flutter_rust_bridge_example_single_block_test` and `flutter_rust_bridge_example_multi_blocks_test` respectively, is that ok?

Reply:
Good idea! LGTM